web/src/message_fetch.js

```diff
@@ -590,83 +512,37 @@ export function initialize(home_view_loaded) {
         //
         // But we do it with a bit of delay, to reduce risk that we
         // hit rate limits with these backfills.
-        const latest_id = data.messages.at(-1).id;
+        const oldest_id = data.messages.at(0).id;
         setTimeout(() => {
             load_messages({
-                anchor: latest_id,
-                num_before: 0,
-                num_after: consts.catch_up_batch_size,
-                msg_list: message_lists.home,
+                anchor: oldest_id,
+                num_before: consts.catch_up_batch_size,
+                num_after: 0,
+                msg_list_data: all_messages_data,
                 cont: load_more,
             });
         }, consts.catch_up_backfill_delay);
     }
 
-    let anchor;
-    if (initial_pointer !== undefined) {
-        // If we're doing a server-initiated reload, similar to a
-        // near: narrow query, we want to select a specific message.
-        anchor = initial_pointer;
-    } else {
-        // Otherwise, we should just use the first unread message in
-        // the user's unmuted history as our anchor.
-        anchor = "first_unread";
-    }
-    load_messages({
-        anchor,
-        num_before: consts.num_before_home_anchor,
-        num_after: consts.num_after_home_anchor,
-        msg_list: message_lists.home,
-        cont: load_more,
+    // Since `all_messages_data` contains continuous message history
+    // with always contains the latest message, it makes sense for
+    // Recent view to display the same data and be in sync.
+    all_messages_data.set_add_messages_callback((messages) => {
+        recent_view_ui.process_messages(messages, all_messages_data);
     });
 
-    if (page_params.is_spectator) {
-        // Since spectators never have old unreads, we can skip the
-        // hacky fetch below for them (which would just waste resources).
-
-        // This optimization requires a bit of duplicated loading
-        // indicator code, here and hiding logic in hide_more.
-        recent_view_ui.show_loading_indicator();
```

Comment:
Is there something we need to do to make loading indicators for the recent view work correctly?